### PR TITLE
[Inductor] Added and_masks and or_masks utilities & make fully masked out rows 0 instead of nan

### PR DESCRIPTION
**Pull request by yanboliang**
Combine #131073 and #131012 and fix doc building failures.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/nn/attention/flex_attention.py [hunk 1]
```diff
@@ -108,13 +117,14 @@ def _identity(
     return score
 
 
-def _no_mask(
+def noop_mask(
```
Comment (joydddd) [inline torch/nn/attention/flex_attention.py:120]: I'm a bit confused... Is `noop_mask` a Tensor or a mask_mod callable?

**Review by Chillee — CHANGES_REQUESTED** [review]
We need a bunch more tests on the nan behavior. cc: @drisspg

File: torch/nn/attention/flex_attention.py (hunk 1 above)
Comment (yanboliang) [reply torch/nn/attention/flex_attention.py:121]: It's the default ```mask_mod``` callable.

File: test/inductor/test_flex_attention.py [hunk 2]
```diff
@@ -199,6 +202,7 @@ def _check_equal(
     ):
         compiled_error = (golden_out - compiled_out).abs().mean()
         ref_error = (golden_out - ref_out).abs().mean()
+        # TODO: Make this check stricter after updating eager SDPA masked_softmax semantics
```
Comment (yanboliang) [inline test/inductor/test_flex_attention.py:205]: After eager SDPA switching to ```_masked_softmax```(#131071), we should enhance it to check no nan for both ```compiled_error``` and ```ref_error```.

**Review by Chillee — APPROVED** [review]

**Pull request closed without merging**